Applications/Spire/Source/Ui/EyeDropperButton.cpp

```diff
@@ -0,0 +1,64 @@
+#include "Spire/Ui/EyeDropperButton.hpp"
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/Ui/Icon.hpp"
+#include "Spire/Ui/Layouts.hpp"
+#include "Spire/Ui/ToggleButton.hpp"
+
+using namespace boost::signals2;
+using namespace Spire;
+using namespace Styles;
+
+EyeDropperButton::EyeDropperButton(QWidget* parent)
+    : QWidget(parent),
+      m_current(std::make_shared<LocalColorModel>()) {
+  m_button = make_icon_toggle_button(
+    imageFromSvg(":/Icons/eye-dropper.svg", scale(16, 16)));
+  m_button->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+  enclose(*this, *m_button);
+  proxy_style(*this, *m_button);
+  setFocusProxy(m_button);
+  m_button->get_current()->connect_update_signal(
+    std::bind_front(&EyeDropperButton::on_checked, this));
+}
+
+const std::shared_ptr<ValueModel<QColor>>&
+    EyeDropperButton::get_current() const {
+  return m_current;
+}
+
+connection EyeDropperButton::connect_submit_signal(
+    const SubmitSignal::slot_type& slot) const {
+  return m_submit_signal.connect(slot);
+}
+
+connection EyeDropperButton::connect_reject_signal(
+    const SubmitSignal::slot_type& slot) const {
+  return m_reject_signal.connect(slot);
+}
+
+void EyeDropperButton::on_checked(bool checked) {
+  if(checked) {
+    auto eye_dropper = new EyeDropper(this);
```

Comment:
How does this object get deleted?

Reply:
`Qt::WA_DeleteOnClose` is set to the `EyeDropper`, so it will be deleted when it is closed.

> Qt::WA_DeleteOnClose Makes Qt delete this widget when the widget has accepted the close event.